479: Uninstall routine

toolstack opened the pull request:
Resolves #479.

toolstack committed:
Add uninstall routine.

Includes a new settings page in WP Admin to allow a user to select to delete the data from the database as well.

Also includes a CLI for deleting the data from the command line.

ocean90 commented on uninstall.php:
```diff
@@ -0,0 +1,52 @@
+	$schema = gp_schema_get();
+
+	foreach ( $schema as $table => $sql ) {
+		$table_name = $wpdb->prefix . 'gp_' . $table;
```
This looks wrong, the prefix can be customized via `$gp_table_prefix`.

ocean90 commented on uninstall.php:
```diff
@@ -0,0 +1,52 @@
+		$wpdb->query( "DROP TABLE {$table_name};" ); // WPCS: unprepared SQL ok.
+	}
+
+	// Delete the WordPress options we use.
```
'WordPress' sounds confusing in this context, the options are used by GlotPress.
Better: `// Delete options.`

There is a lot of 'we' in this function too, this should be avoided.

ocean90 commented on uninstall.php:
```diff
@@ -0,0 +1,52 @@
+	$schema = gp_schema_get();
+
+	foreach ( $schema as $table => $sql ) {
+		$table_name = $wpdb->{'gp_' . $table};
```
Does this work if the plugin is disabled?

toolstack replied:
Yes, that's why the additional defines are included above.

toolstack committed:
Convert settings code to a class.

toolstack committed:
Make sure we have the right table names if the plugin has already been disabled.

toolstack committed:
Add 'IF EXISTS' to the drop table command to avoid errors.

For example if a custom permissions table is defined the default permissions table may already have been deleted and would throw an error.

toolstack committed:
Add $wp_settings to the GP class.

toolstack committed:
Coding standards cleanup.

ocean90 commented on gp-includes/gp.php:
```diff
@@ -165,4 +165,13 @@ class GP {
+	 *
+	 * @var GP_WP_Settings
+	 */
+	public static $wp_settings;
```
This looks weird. Can we make that `$admin` and use it for all (future) stuff which is handled in the WordPress admin?

This should also only be initialized when we're in the admin, not on the front end.

toolstack committed:
Changed GP class variable name and added additional info on the settings page.

Also, only create the instance when in the admin console.

toolstack committed:
Coding standards cleanups.

toolstack committed:
Coding standards cleanups.

toolstack committed:
Update the URL to the glotpress site.
